### DISCUSSION TITLE
Toggle colour mode from Chakra-UI

### DISCUSSION THREAD

**Discussion by SergioArrighi** [rainbow-me/rainbowkit#1163] [Q&A]
Hello all, 
firstly, thanks for this project, it's awesome and I'm loving it <3.
I'm coding a React dApp with Chakra UI.
I've integrated RainbowKit (also following [here](https://www.rainbowkit.com/docs/custom-theme)) and it changes correctly the theme along side the UI when I switch my preferred system theme.
In my app I have also a color mode toggle (as per this [link](https://chakra-ui.com/docs/styled-system/color-mode)), the idea being that sometimes the users may need to switch without modifying the system property.
In fact I didn't find any way to toggle manually the color mode of RainbowKit.
Am I missing something?

Thanks again and keep up the good work!

Have a nice day.

**Comment by Tolu-Mals** [comment]
I'm doing something similar in my project, so I decided to chip in. Chakra UI stores the color mode in `localStorage` or `cookies` with the key `chakra-ui-color-mode`. So you can check for this value and set the theme for RainbowKit based on that. Here's an example of how you could implement dark mode for RainbowKit.

```tsx

const [isDarkTheme, setIsDarkTheme] = useState(false) // light mode by default

...

  useEffect(() => {
    const theme = localStorage.getItem('chakra-ui-color-mode')
    const isDark = theme === 'dark'
    if (isDark) setIsDarkTheme(true)
  }, [])

...

      <RainbowKitProvider
        theme={isDarkTheme ? darkTheme() : lightTheme}
      >
        {children}
      </RainbowKitProvider>

```
